api/users/userModel.js

```diff
@@ -0,0 +1,95 @@
+const db = require('../../data/db-config');
+
+const getAllUsers = async () =>
+  await db('users as u').select(
+    'u.id as user_id',
+    'u.first_name',
+    'u.last_name',
+    'u.school',
+    'u.bg_username',
+    'u.email',
+    'u.phone',
+    'u.created_at'
+  );
+
+const findUserByFilter = async (filter) =>
+  await db('users as u')
+    .where(filter)
+    .select(
+      'u.id as user_id',
+      'u.first_name',
+      'u.last_name',
+      'u.school',
+      'u.bg_username',
+      'u.email',
+      'u.phone',
+      'u.created_at'
+    );
+
+const findUserByID = async (id) =>
+  await db('users as u')
+    .where({ id })
+    .select(
+      'u.id as user_id',
+      'u.first_name',
+      'u.last_name',
+      'u.school',
+      'u.bg_username',
+      'u.email',
+      'u.phone',
+      'u.created_at'
+    )
+    .first();
+
+const createUser = async (user) =>
+  db('users')
+    .insert(user)
+    .returning([
+      'id',
+      'first_name',
+      'last_name',
+      'school',
+      'bg_username',
+      'email',
+      'phone',
+      'created_at',
+    ]);
+
+const updateUser = (id, user) =>
+  db('users')
+    .where({ id })
+    .first()
+    .update(user)
+    .returning([
+      'id',
+      'first_name',
+      'last_name',
+      'school',
+      'bg_username',
+      'email',
+      'phone',
+      'created_at',
+    ]);
+
+const removeUser = async (id) => await db('users').where({ id }).del();
+
+const findOrCreateUser = async (userObj) => {
+  const foundUser = await findUserByID(userObj.id).then((user) => user);
+  if (foundUser) {
+    return foundUser;
+  } else {
+    return await createUser(userObj).then((newUser) => {
+      return newUser ? newUser[0] : newUser;
```

Comment:
Logic here looks a bit confusing. Have you tested it?
Also, why are we returning newUser if newUser is false/undefined? Maybe add some error handling on failure to create?

Reply:
The structure of the code is all provided from the scaffold. I just changed everything from the old Profiles to Users. From what I understand of the code - it's going to return the user being held at `newUser[0]` or just an empty object.